File: experiments/runtime-fields/fixtures/reflective_initial/Entry.java
package probe;

public class Entry {
    public static class Holder { public static Class<?> type = Target.class; }
    public static class Target { public Target() { new Used(); } }
    public static class Used { public Used() { System.out.println("USED"); } }
    public static class Unused {}

    public static void main(String[] args) throws Exception {
        Class<?> type = (Class<?>) Holder.class.getDeclaredField("type").get(null);
        type.getDeclaredConstructor().newInstance();
    }
}

File: experiments/runtime-fields/fixtures/kotlin_object/Entry.kt
package probe

object Holder {
    @JvmField var type: Class<*> = Target::class.java
}

class Target { init { Used() } }
class Used { init { println("USED") } }
class Unused

object Entry {
    @JvmStatic
    fun main(args: Array<String>) {
        val type = Holder::class.java.getDeclaredField("type").get(null) as Class<*>
        type.getDeclaredConstructor().newInstance()
    }
}
